.github/workflows/cache.yml

```diff
@@ -8,52 +8,110 @@ on:
 env:
   GOCACHE: ${{ vars.GOCACHE }}
   GOMODCACHE: ${{ vars.GOMODCACHE }}
+  ASDF_PATHS: |
+    ~/.asdf/installs
+    ~/.asdf/plugins
+    ~/.asdf/shims
+  ELIXIR_PATHS: |
+    platform_umbrella/deps
+    platform_umbrella/_build
+    platform_umbrella/.dialyzer
+    ~/.local/share/bi
+
+  GO_PATHS: |
+    ${{ vars.GOCACHE }}
+    ${{ vars.GOMODCACHE }}
+
+defaults:
+  run:
+    shell: bash
 
 jobs:
-  cache-asdf:
+  check-caches:
```

Comment:
First, check if the cache would be updated. If not, don't run the long-running jobs.